src/aktualizr_primary/secondary.cc

```diff
@@ -141,20 +147,53 @@ class SecondaryWaiter {
   std::unordered_set<std::string> secondaries_to_wait_for_;
 };
 
-static Secondaries createIPSecondaries(const IPSecondariesConfig& config) {
+static Secondaries createIPSecondaries(const IPSecondariesConfig& config, bool provision, Aktualizr& aktualizr) {
```

Comment:
This dependency of the IP Secondary factory on Aktualizr is questionable. Actually, it's me who implicitly introduced this dependency by putting createIPSecondaries() implementation in this file :). Anyway, this is not exactly comment to this PR, just thing that we might need to consider at  some point.